admin-bulk-user-membership/tool/src/main/java/org/sakaiproject/bulk/membership/controller/MainController.java

```diff
@@ -0,0 +1,247 @@
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+* 
+*             http://opensource.org/licenses/ecl2
+* 
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package org.sakaiproject.bulk.membership;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.util.ArrayList;
+import java.util.Locale;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+import org.sakaiproject.bulk.membership.service.BulkUserMembershipToolService;
+import org.sakaiproject.bulk.membership.exception.UsersByEmailException;
+import org.sakaiproject.user.api.User;
+import org.sakaiproject.site.api.Site;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.servlet.LocaleResolver;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+import org.springframework.web.servlet.support.RequestContextUtils;
+
+/**
+ * MainController
+ * 
+ * This is the controller used by Spring MVC to handle requests
+ * 
+ */
+@Slf4j
+@Controller
+public class MainController {
+
+    @Autowired
+    private BulkUserMembershipToolService bulkUserMembershipToolService;
+
+    private final String AUTO_MEMBERSHIP_STEP_1_5 = "auto_membership_step_1_5";
+    private final String AUTO_MEMBERSHIP_STEP_1 = "auto_membership_step_1";
+    private final String AUTO_MEMBERSHIP_STEP_2 = "auto_membership_step_2";
+    private final String AUTO_MEMBERSHIP_STEP_3 = "auto_membership_step_3";
+
+    private ArrayList<String> users = new ArrayList<String>();
+    private String[] roles = null;
+    private ArrayList<Site> sites = new ArrayList<Site>();
+
+    private int currentStep = 1;
+    private String action = "";
+
+    @ModelAttribute("locale")
+	public Locale localeResolver(HttpServletRequest request, HttpServletResponse response) {
+        Locale loc = bulkUserMembershipToolService.getLocaleForCurrentUser();
+        LocaleResolver localeResolver = RequestContextUtils.getLocaleResolver(request);
+        localeResolver.setLocale(request, response, loc);
+        return loc;
+    }
+
+    @GetMapping(value = {"/", "/index"})
+    public String index(Model model) {
+        log.info("Starting the Bulk User Membership tool");
+        startAgain(model);
```

Comment:
Shall we call it resetWizard? startOver?

Reply:
Ok. I will call it resetWizard, I like it